manticore/native/models.py

```diff
@@ -116,33 +116,41 @@ def strcmp(state: State, s1: Union[int, BitVec], s2: Union[int, BitVec]):
     return ret
 
 
-def strlen(state: State, s: Union[int, BitVec]) -> Union[int, BitVec]:
+def strlen(state: State, addr: Union[int, BitVec]) -> Union[int, BitVec]:
     """
-    strlen symbolic model.
+    strlen symbolic model
 
-    Algorithm: Walks from end of string not including NULL building ITE tree when current byte is symbolic.
+    Algorithm: Counts the number of characters in a string forking every time a symbolic byte
+    is found that can be NULL but is not constrained to NULL.
 
-    :param State state: current program state
+    :param state: current program state
     :param s: Address of string
     :return: Symbolic strlen result
-    :rtype: Expression or int
     """
 
-    cpu = state.cpu
+    if issymbolic(addr):
+        raise ConcretizeArgument(state.cpu, 0)
```

Comment:
Have you confirmed that this works as expected? We previously raised `ConcretizeArgument(state.cpu, 1)` here. I thought that the argument index was supposed to be that of the model function, not the _modeled_ function, but I know @ekilmer was working on that recently, so it might have changed.

Reply:
I suggested the change from `1` to `0` based on how we do system call modeling, but now I see they're a little bit different. Will test this.